spec/emerald/parser_spec.rb

```diff
@@ -100,4 +100,48 @@
                           Emerald::Number.new(30)])
     end
   end
+
+  describe "string parsing" do
+    it"parses a string ofcharacters" do
+      source = '"Hello"' 
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"Hello"')])
+    end
+    
+    it"parses a string ofcharacters and whitespace" do
+      source = '"Hello this is a string"' 
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"Hello this is a string"')])
+    end
+
+    it "parses a string of characters, whitespace and digits" do
+      source = '"H3llo this is 1 str1ng"' 
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"H3llo this is 1 str1ng"')])
+    end
+
+    it "parses a string of characters, whitespace, digits and symbols" do
+      source = '"H3llo, this is 1 str1ng."' 
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"H3llo, this is 1 str1ng."')])
+    end
+
+    it "parses unusual strings that contain a backslash immediately followed by a quotation mark" do
+      source = '"this is a weird string -> \" <- "'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"this is a weird string -> \" <- "')])
+    end
+    
+    it "parses strings that contain a backslash immediately followed by a quotation mark" do
+      source = '"try \"this"'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"try \"this"')])
+    end
+
+    it "parses a string with a set of quotation marks are immediately after a backslash" do
+      source = '"Say \"hello\""'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"Say \"hello\""')])
+    end
+  end
 end
```

Comment:
So the extra slashes in the debug output are because Ruby always prints strings with `"`, so you'll see more escapes in it than when you wrote it, if you used a `'` string.

That regex looks good. I'd probably drop the capture group stuff to get `^"([^"\\]|\\.)*"$`

P.S. here's a handy site for Ruby regex testing -> http://rubular.com/r/nMpJJ25EV1